bin/upgrade

```diff
@@ -160,8 +160,25 @@ function handle_image_upgrade() {
   set_server_pro_image_name "$SEED_IMAGE_VERSION"
   docker pull "$IMAGE"
   if [[ $GIT_BRIDGE_ENABLED == "true" ]]; then
-    set_git_bridge_image_name "$SEED_IMAGE_VERSION"
-    docker pull "$GIT_BRIDGE_IMAGE"
+    if [[ -n ${GIT_BRIDGE_IMAGE:-} ]]; then
+      echo "-------------------  WARNING  ----------------------"
+      echo "  You're using a custom git bridge image: $GIT_BRIDGE_IMAGE"
+      echo "  Upgrade of your custom image is skipped. You need to pull the image separately, making sure that:"
+      echo "     1. The Docker image is tagged with the new version: $SEED_IMAGE_VERSION"
+      echo "     2. The environment variable GIT_BRIDGE_IMAGE only contains the image name, and not a tag/version."
+      echo "  You can safely proceed with the upgrade process, but if you try subsequently to restart your instance"
+      echo "  it will fail until the updated image with tag $SEED_IMAGE_VERSION is available in your system"
+      echo "-------------------  WARNING  ----------------------"
+      local should_continue="n"
+      read -r -p "Continue with the upgrade process? [y/n] " should_continue
+      if [[ ! "$should_continue" =~ [Yy] ]]; then
+        echo "Exiting."
+        exit 1
+      fi
+    else
```

Comment:
Do we need to have this in an else branch? Users with custom images will still benefit from pre-pulling the image.

Reply:
If the image is incorrect (have a tag) or it's not in the registry (which I think it's likely unless I'm missing something) it will fail and halt the upgrade process.

If we ask them to upgrade `GIT_BRIDGE_IMAGE` upfront, it will stop working with their current setup if they abort the upgrade. Essentially, we're asking to update `overleaf.rc` before backing it up. I think it's safer to go this way.

Reply:
When they reach the version upgrade, they should have pulled the Toolkit-code update already. So it is safe to ask them to remove the tag from the image at this point.

Reply:
(Sorry, that might not have been clear, we need to move the pulling out of the if-branch :) it's now only pulled when the custom image is set :detective: )

Reply:
Just removed the else branch.